src/deploy/functions/prompts.ts

```diff
@@ -30,7 +30,7 @@ export async function promptForFailurePolicies(
 
   context.existingFunctions =
     (context.existingFunctions as CloudFunctionTrigger[]) ||
-    (await gcp.cloudfunctions.listAllFunctions(context.projectId));
+    (await gcp.cloudfunctions.listAllFunctions(context.projectId)).functions;
```

Comment:
In prepare.ts you only stored the functions.

Reply:
Looking closer, this is probably an unnecessary level of safety - we only use this in one place, and we could easily just pass in a list of existing functions instead of taking them from context.

I'm gonna leave this for now in this PR, and then open another PR to do some of these cleanups

Reply:
Ended up being simpler than expected to just pull this call out entirely, so I did so in this pr

Reply:
::sigh:: If I had a dollar for every time I've had to re-edit prompts.ts while doing this refactor. I'm not sure if it's a code smell that this particular file/test breaks so much.